dynamics/client.py

```diff
@@ -114,11 +117,11 @@ def __init__(
             )
 
         self._api_url = api_url.rstrip("/") + "/"
-        self._session = OAuth2Session(client=BackendApplicationClient(client_id=client_id))
+        self._session = OAuth2Client(client_id, client_secret, scope=scope)
```

Comment:
This attribute should probably be renamed to `_client` since the class initialized is not an `OAuth2Client`.

Reply:
Noted - I went a little more specific and used `_oauth_client`. Hope it works for you.